pallets/rc-migrator/src/accounts.md

```diff
@@ -0,0 +1,218 @@
+sovereign accounts:
+
+- On the Relay: derived from `"para" ++ para_id ++ 00..`
+- On the Asset Hub and all other sibling parachains: derived from `"sibl" ++ para_id ++ 00..`
+
+Our translation logic inverts the derivation and changes the prefix from `"para"` to `"sibl"` for
+all accounts that match the pattern `"para" ++ para_id ++ 00..`. The full list of translated
+accounts is in [this CSV file](./sovereign_account_translation.csv).
+
+It is advised that parachains check that they can control their account on Asset Hub. They can also
+forego this check if they do not need control thereof - for example when they are not holding any
+funds on their relay sovereign account. However, please note that someone could still send funds to
+that address before or after the migration.
+
+Example for Bifrost: this is the [relay sovereign account](https://polkadot.subscan.io/account/13YMK2eeopZtUNpeHnJ1Ws2HqMQG6Ts9PGCZYGyFbSYoZfcm) and it gets translated to this [sibling sovereign account](https://assethub-polkadot.subscan.io/account/13cKp89TtYknbyYnqnF6dWN75q5ZosvFSuqzoEVkUAaNR47A).
+
+## XCM
+
+The migration happens over XCM. There will be events emitted for the balance being removed from the
+Relay Chain and events emitted for the balance being deposited into Asset Hub.
+
+Regular XCM teleport operation, when the Relay Chain acts as the mint authority, follows this
+sequence:
+
+0. Relay Chain: burn_from(source, amount) // emits Balances::Burned event
+0. Relay Chain: mint_into(checking_account, amount) // emits Balances::Minted event  
+0. Relay Chain: total issuance remains unchanged
+0. Relay Chain: sends XCM teleport message
+0. Asset Hub: mint_into(dest, amount) // emits Balances::Minted event
+0. Asset Hub: total issuance increases by `amount`
+
+During account migration, the process will emit `Balances::Burned` events on the Relay Chain and
+`Balances::Minted` events on the Asset Hub. However, unlike regular teleports, it will not mint into
+the `checking_account`, which means the total issuance on the Relay Chain will be updated be the
+migrating amount.
+For more details about the checking account migration and mint authority changes, please refer to
+the sections below.
+
+### Provider and Consumer References
+
+After inspecting the state, it’s clear that fully correcting all reference counts is nearly
+impossible. Some accounts have over `10` provider references, which are difficult to trace and
+reason about. To unwind all of them properly, we would need to analyze the codebase and state
+history, which is not feasible.
+
+Before an account is fully withdrawn from the Relay Chain (RC), we will force-update its consumer
+and provider references to ensure it can be completely removed. If an account is intended to remain
+(fully or partially) on RC, we will update the references accordingly.
+
+To ensure the correct provider and consumer reference counts are established on the Asset Hub (AH),
```

Comment:
@rockbmb Maybe for Polkadot: I think this is a prime bug that we can catch with PET. If we mess reference counts it is very likely that an account cannot destroy itself because it has a consumer left. 

We can have a "smart" test that looks at all accounts, and try and "unwind" all of them towards destruction. It would have to inspect that the account has consumers in what pallets (proxy, multisig, staking, opengov etc) and remove all associated consumers by removing the account from the said pallets. Finally, we should assert that the account is in a state that it can perform "transfer_all" and kill itself.